cdap-ui/app/cdap/components/TimeRangePicker/index.js

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Calendar from 'react-calendar';
+import moment from 'moment';
+import classnames from 'classnames';
+
+require('./TimeRangePicker.scss');
+
+const format = 'MM-DD-YYYY HH:mm A';
+
+export default class TimeRangePicker extends Component {
+  static propTypes = {
+    onChange: PropTypes.func,
+    start: PropTypes.number,
+    end: PropTypes.number
+  };
+
+  state = {
+    date: null,
+    displayCalendar: 'start',
+    hour: 0,
+    minute: 0,
+    start: this.props.start,
+    end: this.props.end,
+  };
+
+  changeDate = (date) => {
+    this.setState({
+      date,
+      [this.state.displayCalendar]: this.calculateDisplayTime(date)
+    }, () => {
+      this.applyChange();
+    });
+  };
+
+  changeDisplay = (type) => {
+    if (type === this.state.displayCalendar) { return; }
+
+    let date = this.state.date,
+        hour = this.state.hour,
+        minute = this.state.minute;
+
+    if (this.state[type]) {
+      let time = new Date(this.state[type]);
+      date = time;
+      hour = time.getHours();
+      minute = time.getMinutes();
+    }
+
+    this.setState({
+      displayCalendar: type,
+      date,
+      hour,
+      minute
+    });
+  };
+
+  changeHour = (e) => {
+    let hour = e.target.value;
+
+    this.setState({
+      hour,
+      [this.state.displayCalendar]: this.calculateDisplayTime(undefined, hour)
```

Comment:
nit: can we pass `null` instead of `undefined`?

Reply:
no, we can't. `null` is still a valid value, so the default on the function argument will not apply.

Reply:
though this is probably not good practice... i'll change the function signature.. lol